### PR TITLE
Update scheduler to use schedulerName selector

### LINKED ISSUE #52254: Enhance multi-scheduler to use schedulerName selector.
**Is this a BUG REPORT or FEATURE REQUEST?**:
After https://github.com/kubernetes/kubernetes/pull/50582 , we can select pods for multi-scheduler. PS: if pod did not set schedulerName, 'default' will be set as default value.

/kind cleanup

@kubernetes/sig-scheduling-misc

### PR DESCRIPTION
**Pull request by wanghaoran1988**
**What this PR does / why we need it**:
Update scheduler to use schedulerName selector when select pods in the podInformer
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #52254

**Special notes for your reviewer**:

**Release note**:

```
None
```

### CODE REVIEW

File: plugin/pkg/scheduler/factory/factory.go [hunk 1]
```diff
@@ -894,8 +894,8 @@ func (i *podInformer) Lister() corelisters.PodLister {
 }
 
 // NewPodInformer creates a shared index informer that returns only non-terminal pods.
-func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) coreinformers.PodInformer {
-	selector := fields.ParseSelectorOrDie("status.phase!=" + string(v1.PodSucceeded) + ",status.phase!=" + string(v1.PodFailed))
+func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration, schedulerName string) coreinformers.PodInformer {
+	selector := fields.ParseSelectorOrDie("spec.schedulerName=" + schedulerName + ",status.phase!=" + string(v1.PodSucceeded) + ",status.phase!=" + string(v1.PodFailed))
```
Comment (resouer) [inline plugin/pkg/scheduler/factory/factory.go]: Please make this line shorter

**Comment by k82cn** [comment]
with `schedulerName`, `ConfigFactory.ResponsibleForPod` is not necessary.

**Comment by k82cn** [comment]
BTW, please also remove related test for `ResponsibleForPod`.

File: plugin/pkg/scheduler/factory/factory.go (hunk 1 above)
Comment (dixudx) [reply plugin/pkg/scheduler/factory/factory.go]: +1. Can use `fmt.Sprintf` or `strings.Join()` to make it more concise and readable.

**Comment by wanghaoran1988** [comment]
Updated, `ResponsibleForPod` was removed and updated the integration testing `TestMultiScheduler`

**Comment by wanghaoran1988** [comment]
@k82cn @resouer PTAL

File: plugin/pkg/scheduler/factory/factory_test.go [hunk 2]
```diff
@@ -357,95 +357,6 @@ func TestBind(t *testing.T) {
 	}
 }
 
-// TestResponsibleForPod tests if a pod with an annotation that should cause it to
-// be picked up by the default scheduler, is in fact picked by the default scheduler
-// Two schedulers are made in the test: one is default scheduler and other scheduler
-// is of name "foo-scheduler". A pod must be picked up by at most one of the two
-// schedulers.
-func TestResponsibleForPod(t *testing.T) {
```
Comment (k82cn) [inline plugin/pkg/scheduler/factory/factory_test.go]: Is it possible to add other integration or e2e test case for multi-scheduler?

**Comment by k82cn** [comment]
The functional part is OK to me; need to add test for the change.

**Comment by uptoknow** [comment]
@k82cn It's covered in the TestMultiScheduler here: 
https://github.com/kubernetes/kubernetes/pull/52479/files#diff-cdc3caeca8898a0c12df164e679baef3R421

**Comment by k82cn** [comment]
LGTM for the scheduler part; so how, I'm not sure the performance impact to apiserver (sorry for that, did not highlight when opened the issue); if it's just a piece of logic for `schedulerName` (together with other condition), that'll be great.

**Comment by wanghaoran1988** [comment]
@liggitt @deads2k Sorry to bother you, but do you know whether select pods by the fieldSlector spec.SchedulerName have performance issue to the apiServer? I am quite not sure how api server build the select index for etcd.

**Comment by liggitt** [comment]
there is no etcd index. all fieldSelector filtering (other than specific exceptions like namespace and name) is done in the apiserver.

**Comment by davidopp** [comment]
Why did you delete a large test?

**Comment by wanghaoran1988** [comment]
@davidopp See comments here: https://github.com/kubernetes/kubernetes/pull/52479#issuecomment-329468915

**Comment by davidopp** [comment]
Thanks. But it looks like you did not add a new test like @k82cn was suggesting?

**Comment by wanghaoran1988** [comment]
@davidopp I updated the exits multiple scheduler test, it already covered this.

**Comment by k82cn** [comment]
LGTM, overall :). Can you help to rebase that.

/cc @bsalamat

**Review by bsalamat — COMMENTED** [review]
Looks like TestMultiScheduler in integration/scheduler/scheduler_test.go covers the changes here. So, it LGTM.

/lgtm

**Pull request merged**